pycbc/inference/sampler/emcee_pt.py

```diff
@@ -171,6 +172,14 @@ def from_config(cls, cp, model, nprocesses=1, use_mpi=False):
         obj.set_burn_in_from_config(cp)
         # set prethin options
         obj.set_thin_interval_from_config(cp, section)
+        # Set up the output file
+        setup_output(obj, output_file)
+        if not obj.new_checkpoint:
+            obj.resume_from_checkpoint()
+        else:
+            init_prior = initial_dist_from_config(cp, obj.variable_params)
+            obj.set_p0(prior=init_prior)
+            obj._lastclear = 0
```

Comment:
To please code climate, and to reduce copying code, create a function in `BaseMCMC` called something like `set_start_from_config` that does these three lines of code:
```
def set_start_from_config(self, cp):
    init_prior = initial_dist_from_config(cp, self.variable_params)
    self.set_p0(prior=init_prior)
```
(Don't do the `_lastclear` here, since you'll be putting that back into the beginning of the `run` function.)

Then call `self.set_start_from_config(cp)` here.